mesoshttp/client.py

```diff
@@ -574,12 +628,13 @@ def __register(self):
         if self.failover_timeout:
             subscribe['subscribe']['framework_info']['failover_timeout'] = self.failover_timeout
 
-        if self.authenticate:
```

Comment:
It didn't make sense to me to need `authenticate`, `principal`, and `secret`, when you can simply check them serially for being not `None`. I have no strong opinion in either direction, though. This just seemed like a good, purely internal improvement.

Reply:
correct